test/test_using_plams/test_cp2k_mm.py

```diff
@@ -0,0 +1,176 @@
+            if '[FREQ]' in item:
+                break
+        else:
+            raise ValueError(f"failed to identify the '[FREQ]' substring in {f!r}")
+
+        # Identify the vibrational degrees of freedom
+        if not atom_count:
+            raise ValueError(f"failed to identify any atoms in the '[Atoms]' section of {f!r}")
+        elif atom_count in {1, 2}:
+            count = atom_count - 1
+        else:
+            count = 3 * atom_count - 6
+
+        # Gather and return the frequencies
+        iterator = islice(f, 0, count)
+        ret = np.fromiter(iterator, dtype=float, count=count)
+        ret *= Units.conversion_ratio('cm-1', unit)
+        return ret
+
+
+def overlap_coords(xyz1: np.ndarray, xyz2: np.ndarray) -> np.ndarray:
+    """Rotate *xyz1* such that it overlaps with *xyz2* using the Kabsch algorithm."""
+    xyz1 = np.asarray(xyz1, dtype=float)
+    xyz2 = np.asarray(xyz2, dtype=float)
+
+    # Peform a singular value decomposition on the covariance matrix
+    H = xyz1.T @ xyz2
+    U, _, Vt = np.linalg.svd(H)
+    V, Ut = Vt.T, U.T
+
+    # Construct the rotation matrix
+    rotmat = np.eye(3)
+    rotmat[2, 2] = np.linalg.det(V @ Ut)
+    rotmat = V @ rotmat @ Ut
+
+    return xyz1 @ rotmat.T
+
+
+@add_to_class(Cp2kJob)
+def get_runscript(self):
+    inp, out = self._filename('inp'), self._filename('out')
+    return f'cp2k.ssmp -i {inp} -o {out}'
+
+
+def get_energy(self, index: int = -1, unit: str = 'Hartree') -> float:
+    """Return the energy of the last occurence of ``'ENERGY| Total FORCE_EVAL'`` in the output."""
+    energy_str = self.grep_output('ENERGY| Total FORCE_EVAL')[index]
+    energy = float(energy_str.rsplit(maxsplit=1)[1])
+    return Units.convert(energy, 'Hartree', unit)
+
+
+@pytest.mark.slow
+@delete_output(delete_workdir=True)
+def test_singlepoint() -> None:
+    """Test CP2K singlepoint calculations with the :class:`CP2K_MM` class."""
+    s = SETTINGS.copy()
+    s.specific.cp2k += singlepoint.specific.cp2k_mm
+
+    job = cp2k_mm(settings=s, mol=MOL)
+    result = run(job, path=PATH)
+    assertion.eq(result.status, 'successful')
+
+    # Yes, this is a small hack as neither energy nor get_energy() seems to work
+    plams_results = result.results
+    add_to_instance(plams_results)(get_energy)
+    setattr(result, 'get_energy', plams_results.get_energy)
+
+    # Compare energies
+    ref = -15.4431781758
+    energy = result.get_energy()
+    assertion.isclose(energy, ref, rel_tol=10**-4)
+
+
+@pytest.mark.slow
+@delete_output(delete_workdir=True)
+def test_geometry() -> None:
+    """Test CP2K geometry optimization calculations with the :class:`CP2K_MM` class."""
+    s = SETTINGS.copy()
+    s.specific.cp2k += geometry.specific.cp2k_mm
+
+    job = cp2k_mm(settings=s, mol=MOL)
+    result = run(job, path=PATH)
+    assertion.eq(result.status, 'successful')
+
+    # Yes, this is a small hack as neither energy nor get_energy() seems to work
```

Comment:
Do you know why it is not working?

Reply:
Just like the ``Package`` properties it searches for ``"Total energy"``, a line which is absent from the output of these MM calculations. I suspect that might be an output specific to DFT. (https://github.com/SCM-NV/qmflows/blob/bb40efe534b54b742d71229ca77227f38f7ac907/src/qmflows/data/dictionaries/propertiesCP2K.yaml#L1-L7)